src/EyeTracker.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect } from 'react';
+import './App.css'
+
+const EyeTracker = () => {
+    const [clickCount, setClickCount] = useState([0, 0, 0, 0, 0]);
+    const [totalClicks, setTotalClicks] = useState(0);
+    const NUMBER_OF_DOTS = 5;
+    const MAX_CLICKS = 12;
+    const MAX_TOTAL_CLICKS = MAX_CLICKS * NUMBER_OF_DOTS;
+
+    let xprediction : number;
+    let yprediction : number;
+
+    useEffect(() => {
+        webgazer.setGazeListener(function(data, elapsedTime) {
+            if (data == null) {
+                return;
+            }
+            xprediction = data.x; //these x coordinates are relative to the viewport
+            yprediction = data.y; //these y coordinates are relative to the viewport
+            console.log(elapsedTime); //elapsed time is based on time since begin was called
+        }).begin();
+      })
+
+
+    const handleClick = (index: number) => {
+        const newClickCount = [...clickCount];
+        newClickCount[index]++;
+        setClickCount(newClickCount);
+        setTotalClicks(totalClicks + 1);
+    };
+
+    const getVisibility = () => {
+        if (totalClicks >= MAX_TOTAL_CLICKS) {
+            return 'hidden';
+        }   
+        else {
+            return 'visible';
+        }
+    };
+
+    const getDotColor = (index: number) => {
+        if (clickCount[index] >= 12) {
```

Comment:
Please avoid [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)). What is 12? Create a constant for it.